### PR TITLE
configs/imxrt1050-evk: Add tc-protected config

### PR DESCRIPTION
**Pull request by yashwanth686007**
This patch adds defconfig for kernel & driver testcases on imxrt1050-evk protected build.

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [cbcda59ce7e3101d21248f124428e2b1fd852d03] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [cbcda59ce7e3101d21248f124428e2b1fd852d03] - Code Rule Check OK.

File: apps/examples/testcase/le_tc/kernel/tc_libc_misc.c [hunk 1]
```diff
@@ -227,6 +227,7 @@ static void tc_libc_misc_crc32part(void)
 	TC_SUCCESS_RESULT();
 }
 
+#ifndef CONFIG_BUILD_PROTECTED
```
[the remark this answers is not in the document]
Comment (yashwanth686007) [reply apps/examples/testcase/le_tc/kernel/tc_libc_misc.c]: Yes, there seems to be some issue with the debug commands. I will update it later. Rightnow, I have reverted the commit.

**Comment by seinfra** [comment]
Target : [a530e9ab15209e285af6c5d3859fb59a8b772ee5] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [a530e9ab15209e285af6c5d3859fb59a8b772ee5] - Code Rule Check OK.

**Pull request merged**